### DISCUSSION TITLE
Vercel Edge functions, Serverless functions and nextjs api routes or proper backend?

### DISCUSSION THREAD

**Discussion by jlorezz** [vercel/community#504] [Help] [answered]
What are the benefits of using vercel edge functions or serverless functions compared to a proper backend (for example nestjs)?
or maybe its better to put it this way? At what point should we consider going for a proper backend and not just nextjs api routes?

**Comment by ismaelrumzan** [comment, marked as answer]
Hi there. Thanks for your question!
One of the key benefits of serverless and edge functions is that you will benefit from edge cache available in multiple regions.
I would suggest reviewing this [page](https://vercel.com/docs/concepts/functions/conceptual-model) that describes the conceptual model of how they work as well as this [page](https://vercel.com/docs/concepts/functions/edge-caching), that explains how edge cache works with serverless functions

**Reply by jlorezz** [reply]
awesome thanks!
